scripts/new-version-pr-generator.js

```diff
@@ -0,0 +1,180 @@
+/* eslint-disable no-console */
+
+const {execSync} = require('child_process');
+const {readFileSync} = require('fs');
+const pathResolve = require('path').resolve;
+const {mkdir, rm} = require('shelljs');
+const yaml = require('js-yaml');
+const semver = require('semver');
+
+const {version: PACKAGE_VERSION} = require('../package.json');
+const secrets = require('../secrets.json');
+const Retry = require('./utilities/retry');
+
+const YARN_VERSION = yaml
+  .safeLoad(readFileSync(pathResolve(__dirname, '..', 'dev.yml'), 'utf8'))
+  .up.find((obj) => Object.keys(obj).includes('node')).node.yarn;
+
+// File paths and directories
```

Comment:
Can you all the comments in this file? We prefer to write self-evident code, and only comment when there's something really unusual going on. Comments like the ones in this file actually make the code less readable by breaking the logical flow and being visual noise